### PR TITLE
Server installation

### PR DESCRIPTION
**Pull request by jburel**
Review installation instructions

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-docs-superbuild-push#25](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-docs-superbuild-push/25/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-docs-superbuild-push/25/consoleText) for more details.
Possible conflicts:
  - PR #2035 joshmoore 'Merge build (WIP)'
    - omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh

~~--conflicts~~ Conflict resolved in build [OMERO-docs-superbuild-push#26](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-docs-superbuild-push/26/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-docs-superbuild-push/26/consoleText) for more details.

**Comment by joshmoore** [comment]
My fault, @jburel. I accidentally included files that were code-generated. They've now been removed.

**Comment by jburel** [comment]
Thanks

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-docs-superbuild-push#28](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-docs-superbuild-push/28/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-docs-superbuild-push/28/consoleText) for more details.
Possible conflicts:
  - PR #2035 joshmoore 'Merge build (WIP)'
    - omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh

--conflicts

**Comment by jburel** [comment]
Following today's discussion, the walkthroughs will need to be reviewed.
If we are happy with the other changes it might be preferable to ignore changes made in the walkthroughs  in that PR
and rework them in a follow-up one

File: omero/sysadmins/server-advanced-configuration.rst [hunk 1]
```diff
@@ -129,18 +129,18 @@ To run multiple servers on a single host, the easiest approach is to prefix all
 ports (SSL, TCP, registry) using :property:`omero.ports.prefix`::
 
     # First server
-    cd ~/OMERO.server-1
-    bin/omero admin start
+    OMERODIR = ~/OMERO.server-1
```
Comment (manics) [inline omero/sysadmins/server-advanced-configuration.rst]:
```suggestion
    export OMERODIR=~/OMERO.server-1
```
You can't have spaces around the `=` when defining a shell environment variable. I think `export` is required (I haven't tested this, we should schedule a full test of all the docs before the RC release)

File: omero/sysadmins/server-advanced-configuration.rst [hunk 2]
```diff
@@ -129,18 +129,18 @@ To run multiple servers on a single host, the easiest approach is to prefix all
...
+    omero admin start
 
     # Second server
-    cd ~/OMERO.server-2
-    bin/omero config set omero.ports.prefix 1
-    bin/omero admin start
+    OMERODIR = ~/OMERO.server-2
```
Comment (manics) [inline omero/sysadmins/server-advanced-configuration.rst]:
```suggestion
    export OMERODIR=~/OMERO.server-2
```

File: omero/sysadmins/server-advanced-configuration.rst [hunk 3]
```diff
@@ -129,18 +129,18 @@ To run multiple servers on a single host, the easiest approach is to prefix all
...
+    omero config set omero.ports.prefix 1
+    omero admin start
 
     # Third server
-    cd ~/OMERO.server-3
-    bin/omero config set omero.ports.prefix 2
-    bin/omero admin start
+    OMERODIR = ~/OMERO.server-3
```
Comment (manics) [inline omero/sysadmins/server-advanced-configuration.rst]:
```suggestion
    export OMERODIR=~/OMERO.server-3
```

File: omero/sysadmins/server-tables.rst [hunk 4]
```diff
@@ -90,7 +90,7 @@ After that, the following should succeed:
   your operating system's package installer if possible or if you must
   instead use PyPI_ directly::
 
-    pip install mock
+    pip3 install mock
```
Comment (manics) [inline omero/sysadmins/server-tables.rst]: FYI need to be careful with `pip` / `pip3`. Sometimes the `3` version is missing from a virtualenv in which case `pip3` might refer to a global pip. Something to check later.

File: omero/sysadmins/unix/install-web/web-deployment.rst [hunk 5]
```diff
@@ -142,17 +104,32 @@ If possible, install the following packages:
     * - System
       - Package
 
-    * - BSD Ports
-      - lang/python27 graphics/py-pillow math/py-matplotlib math/py-numpy www/nginx
-
     * - Debian
-      - python2.7 python-pil python-matplotlib python-numpy nginx
+      - python3 python3-numpy nginx
```
Comment (manics) [inline omero/sysadmins/unix/install-web/web-deployment.rst]: If everything is in a virtualenv `python3-numpy` isn't needed
```suggestion
      - python3 nginx
```

File: omero/sysadmins/unix/install-web/web-deployment.rst [hunk 6]
```diff
@@ -142,17 +104,32 @@ If possible, install the following packages:
...
 
     * - Homebrew
-      - python pillow numpy matplotlib nginx
+      - python numpy nginx
```
Comment (manics) [inline omero/sysadmins/unix/install-web/web-deployment.rst]:
```suggestion
      - python nginx
```

File: omero/sysadmins/unix/install-web/web-deployment.rst [hunk 7]
```diff
@@ -142,17 +104,32 @@ If possible, install the following packages:
...
 
     * - RedHat
-      - python nginx
+      - python3 numpy nginx
```
Comment (manics) [inline omero/sysadmins/unix/install-web/web-deployment.rst]:
```suggestion
      - python3 nginx
```

File: omero/sysadmins/unix/install-web/web-deployment.rst [hunk 8]
```diff
@@ -142,17 +104,32 @@ If possible, install the following packages:
...
+Installation
+------------
+
+From **OMERO 5.6.0** release, the ``omero-web`` library supports Python 3 and
+can be installed via :command:`pip`.
+
+We assume the following::
+
+    $ omero_user_home_dir=/home/omero
+    $ OMERODIR=${omero_user_home_dir}/omero
```
Comment (manics) [inline omero/sysadmins/unix/install-web/web-deployment.rst]:
```suggestion
    $ export OMERODIR=${omero_user_home_dir}/omero
```

File: omero/sysadmins/unix/install-web/web-deployment.rst [hunk 9]
```diff
@@ -142,17 +104,32 @@ If possible, install the following packages:
...
+
+We recommend you use a virtual environment::
+
+    $ python3 -m venv py3_venv
+    $ source py3_venv/bin/activate
+    $ pip install omero-web
+    $ OMERODIR=${OMERODIR}
```
Comment (manics) [inline omero/sysadmins/unix/install-web/web-deployment.rst:132]: What's the intention of this line?

File: omero/sysadmins/unix/install-web/web-deployment.rst [hunk 10]
```diff
@@ -200,7 +177,7 @@ a minimal configuration:
 
 ::
 
-    $ OMERO.py/bin/omero web config nginx-location > /home/omero/omero-web-location.include
+    $ omero web config nginx-location > {{ omero_user_home_dir }}/omero-web-location.include
```
Comment (manics) [inline omero/sysadmins/unix/install-web/web-deployment.rst]:
```suggestion
    $ omero web config nginx-location > ${omero_user_home_dir}/omero-web-location.include
```

File: omero/sysadmins/unix/server-installation.rst [hunk 11]
```diff
@@ -185,17 +184,14 @@ If possible, install the following packages:
     * - System
       - Package
 
-    * - BSD Ports
-      - lang/python27 math/py-numpy devel/py-tables
-
     * - Debian
-      - python2.7 python-numpy python-tables
+      - python3 python3-numpy
```
Comment (manics) [inline omero/sysadmins/unix/server-installation.rst]:
```suggestion
      - python3
```

File: omero/sysadmins/unix/server-installation.rst [hunk 12]
```diff
@@ -185,17 +184,14 @@ If possible, install the following packages:
...
 
     * - Homebrew
-      - python2 numpy
+      - python3 numpy
```
Comment (manics) [inline omero/sysadmins/unix/server-installation.rst]:
```suggestion
      - python3
```

File: omero/sysadmins/unix/server-installation.rst [hunk 13]
```diff
@@ -185,17 +184,14 @@ If possible, install the following packages:
...
 
     * - RedHat
-      - python
+      - python3 numpy
```
Comment (manics) [inline omero/sysadmins/unix/server-installation.rst]:
```suggestion
      - python3
```

File: omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh [hunk 14]
```diff
@@ -15,11 +15,11 @@ yum -y install java-1.8.0-openjdk
 
 # install dependencies
 
-yum -y install python-{pip,devel,virtualenv,yaml,jinja2,tables}
+yum -y install python3-{pip,devel,virtualenv,yaml,jinja2,tables}
```
Comment (manics) [inline omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh:18]:
```suggestion
yum -y install python3
```

File: omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh [hunk 15]
```diff
@@ -30,8 +30,6 @@ yum -y install libdb-utils
 yum -y install openssl-devel bzip2-devel
```
Comment (manics) [inline omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh:30]: I don't think these are necessary if everything is installed using wheels

File: omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh [hunk 16]
```diff
@@ -66,32 +64,48 @@ su - postgres -c "createdb -E UTF8 -O '$OMERO_DB_USER' '$OMERO_DB_NAME'"
...
 unzip -q OMERO.server*
 #end-release-ice36
 ln -s OMERO.server-*/ OMERO.server
-OMERO.server/bin/omero config set omero.data.dir "$OMERO_DATA_DIR"
-OMERO.server/bin/omero config set omero.db.name "$OMERO_DB_NAME"
-OMERO.server/bin/omero config set omero.db.user "$OMERO_DB_USER"
-OMERO.server/bin/omero config set omero.db.pass "$OMERO_DB_PASS"
-OMERO.server/bin/omero db script -f OMERO.server/db.sql --password "$OMERO_ROOT_PASS"
+# set OMERODIR
+export OMERODIR=OMERO.server
```
Comment (manics) [inline omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh:90]: probably safer to use the full absolute path

File: omero/sysadmins/unix/walkthrough/walkthrough_debian9.sh [hunk 17]
```diff
@@ -18,24 +18,23 @@ apt-get -y install openjdk-8-jre-headless
 
 # install dependencies
 
-apt-get -y install python-{pip,virtualenv,yaml,jinja2}
+apt-get -y install python3-{pip,venv,yaml,jinja2}
```
Comment (manics) [inline omero/sysadmins/unix/walkthrough/walkthrough_debian9.sh:21]:
```suggestion
apt-get -y install python3-{pip,venv}
```

**Review by manics — COMMENTED** [review]
Looks generally fine.

Just realised I've commented on some of the auto-generated docs...

File: omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh (hunk 14 above)
Comment (jburel) [reply omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh:18]: Those changes are made in omero-install so they will be overridden

**Comment by jburel** [comment]
@manics changes to the walkthroughs have to be done in omero-install.
I am currently working on that. so no need to integrate them

File: omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh (hunk 16 above)
Comment (jburel) [reply omero/sysadmins/unix/walkthrough/walkthrough_centos7.sh:90]: this has been changed in omero-install

File: omero/sysadmins/server-advanced-configuration.rst (hunk 1 above)
Comment (jburel) [reply omero/sysadmins/server-advanced-configuration.rst]: space added by mistake

**Review by manics — APPROVED** [review]
As discussed in the office merge this and continue working in separate PRs.

**Pull request merged**